junit/MatrixTest.java

```diff
@@ -362,6 +361,83 @@ public void testGetException()
         testIndex[1] = -1; 
         matrix.get(testIndex[0], testIndex[1]);
     }
+    /**
+     * testScalarMult -Test for the scalar times function.
+     */
+    @Test
+    public void testScalarMult() 
+    {
+        Random rand = new Random();
+        int multiplier = rand.nextInt(100);
+        double[][] testMatrix = {
+            {1, 2, 3, 4},
+            {5, 6, 7, 8},
+            {9, 10, 11, 12}
+        };
+        double[][] corrMatrix = new 
+            double[testMatrix.length][testMatrix[0].length];
+        for (int i = 0; i < testMatrix.length; i++) 
+        {
+            for (int j = 0; j < testMatrix[i].length; j++) 
+            {
+                corrMatrix[i][j] = testMatrix[i][j] * multiplier;
+            }
+        }
+        Matrix myMatrix = new Matrix(testMatrix);
+        Matrix matrixResult = myMatrix.times(multiplier);
+        assertEquals(corrMatrix, matrixResult.getArray());
+    }
+    /**
+     * testTimesEquals -Tests the arrayTimesEquals method,
+     *  element wise multiplication of a scalar.
+     */
+    @Test
+    public void testTimesEquals()
+    {
+        Random rand = new Random();
+        double multiplier = rand.nextDouble(); 
+        double[][] testMatrix = {
+            {1, 2, 3, 4},
+            {5, 6, 7, 8},
+            {9, 10, 11, 12}
+        };
+        Matrix testMatrixA = new Matrix(testMatrix);
+        for (int i = 0; i < testMatrix.length; i++)
+        {
+            for (int j = 0; j < testMatrix[i].length; j++)
+            {
+                testMatrix[i][j] = testMatrix[i][j] * multiplier;
+            }
+        }
+        testMatrixA.timesEquals(multiplier);
+        assertEquals(testMatrixA.getArray(), testMatrix);
+    }
+    /**
+     * testDotProduct -Tests the dot product multiplication of
+     *  matrices.
+     */
+    @Test
+    public void testDotProduct() 
```

Comment:
From what I can see, this is actually testing normal matrix multiplication. Is that correct? If so, how can we be multiplying that 3x4 matrix by itself? Isn't that illegal. I think I'm missing something here.